daemon/api.go

```diff
@@ -731,20 +731,25 @@ func (*licenseData) Error() string {
 	return "license agreement required"
 }
 
+type snapDownloadOptions struct {
+	Channel   string        `json:"channel"`
+	Revision  snap.Revision `json:"revision"`
+	CohortKey string        `json:"cohort-key"`
+}
+
 type snapInstruction struct {
 	progress.NullMeter
-	Action           string        `json:"action"`
-	Amend            bool          `json:"amend"`
-	Channel          string        `json:"channel"`
-	Revision         snap.Revision `json:"revision"`
-	CohortKey        string        `json:"cohort-key"`
-	LeaveCohort      bool          `json:"leave-cohort"`
-	DevMode          bool          `json:"devmode"`
-	JailMode         bool          `json:"jailmode"`
-	Classic          bool          `json:"classic"`
-	IgnoreValidation bool          `json:"ignore-validation"`
-	Unaliased        bool          `json:"unaliased"`
-	Purge            bool          `json:"purge,omitempty"`
+	snapDownloadOptions
```

Comment:
Then place those just where Revision etc was in the struct.  It's strange for this not to lead with Action.

Reply:
Indeed, thanks! Fixed as well.